examples/training/textual_inversion.py

```diff
@@ -0,0 +1,787 @@
+                lr_scheduler.step()
+                optimizer.zero_grad()
+
+            # Checks if the accelerator has performed an optimization step behind the scenes
+            if accelerator.sync_gradients:
+                progress_bar.update(1)
+                global_step += 1
+
+            logs = {"loss": loss.detach().item(), "lr": lr_scheduler.get_last_lr()[0], "step": global_step}
+            progress_bar.set_postfix(**logs)
+            accelerator.log(logs, step=global_step)
+
+            if global_step >= args.max_train_steps:
+                break
+        progress_bar.close()
+
+        accelerator.wait_for_everyone()
+
+        # Generate sample images for visual inspection
+        # if accelerator.is_main_process:
+        #     if epoch % args.save_images_epochs == 0 or epoch == args.num_epochs - 1:
+        #         pipeline = StableDiffusionPipeline(
+        #             unet=accelerator.unwrap_model(stable_diffusion.unet),
+        #             vae=accelerator.unwrap_model(stable_diffusion.vae),
+        #             text_encoder=accelerator.unwrap_model(stable_diffusion.text_encoder),
+        #             tokenizer=tokenizer,
+        #             scheduler=noise_scheduler,
+        #             safety_checker=StableDiffusionSafetyChecker.from_pretrained("CompVis/stable-diffusion-safety-checker"),
+        #             feature_extractor=CLIPFeatureExtractor.from_pretrained("openai/clip-vit-base-patch32")
+        #         )
+
+        #         generator = torch.manual_seed(0)
+        #         # run pipeline in inference (sample random noise and denoise)
+        #         images = pipeline(generator=generator, batch_size=args.eval_batch_size, output_type="numpy")["sample"]
+
+        #         # denormalize the images and save to tensorboard
+        #         images_processed = (images * 255).round().astype("uint8")
+        #         accelerator.trackers[0].writer.add_images(
+        #             "test_samples", images_processed.transpose(0, 3, 1, 2), epoch
+        #         )
+
+        #     if epoch % args.save_model_epochs == 0 or epoch == args.num_epochs - 1:
+        #         # save the model
+        #         if args.push_to_hub:
+        #             push_to_hub(args, pipeline, repo, commit_message=f"Epoch {epoch}", blocking=False)
+        #         else:
+        #             pipeline.save_pretrained(args.output_dir)
+        accelerator.wait_for_everyone()
+
+    # Create the pipeline using using the trained modules and save it.
+    if accelerator.is_main_process:
+        # Load the CLIPTextualInversionWrapper into CLIPTextModel.
+        text_model = accelerator.unwrap_model(model.text_encoder)
+        # Update the token embeddings of the placeholder_token to the trained embeddings.
+        text_model.merge_concept_embeddings_in_token_embeddings()
+        clip = CLIPTextModel(text_model.config)
+        clip.load_state_dict(text_model.state_dict(), strict=False)
```

Comment:
Here we merge the trained embeddings in  token embeddings and load the weights in CLIP to save with `StableDiffusionPipeline`